v23.1/online-schema-changes.md

```diff
@@ -93,6 +95,65 @@ Declarative schema changer statements and legacy schema changer statements opera
 
 ## Best practices for online schema changes
 
+### Estimate your storage capacity before performing online schema changes
+
+Some schema change operations like adding or dropping columns, or altering primary keys will result in temporary increases to a cluster's storage consumption. These operations will result in consuming at least three times the storage space of index while the schema change is being applied, which may cause the cluster to run out of storage space and pause or fail to apply the schema change.
+
+To estimate the size of the indexes in your table, use the [`SHOW RANGES`](show-ranges.html) statement.
+
+{% include_cached copy-clipboard.html %}
+~~~ shell
+SHOW RANGES FROM TABLE {table name} WITH DETAILS, KEYS, INDEXES;
+~~~
+
+The output includes a `range_size_mb` column that shows the size of the range in megabytes for each index.
+
+In many cases this range size is trivial, but when the range size is many gigabytes or terabytes, you will need at least three times that amount of free storage space to successfully apply an online schema change.
+
+#### Example of finding the range size of an index
+
+1. Start a 3 node [`cockroach demo`](cockroach-demo.html) cluster with the MovR dataset.
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    cockroach demo --nodes 3
+    ~~~
+
+1. Turn off the deprecated behavior of `SHOW RANGES`:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ sql
+    SET CLUSTER SETTING sql.show_ranges_deprecated_behavior.enabled TO 'false';
+    ~~~
+
+1. Find the range size of the indexes in the `movr.vehicles` table:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ sql
+    WITH x AS (
+            SHOW RANGES FROM TABLE movr.vehicles WITH DETAILS, KEYS, INDEXES
+         )
+    SELECT index_name,
+           raw_index_start_key,
```

Comment:
Maybe make this more user-friendly and let's not select keys

Reply:
Done.